bottlerocket-settings-models/settings-extensions/bootstrap-commands/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+//! Settings related to bootstrap commands.
+use bottlerocket_model_derive::model;
+use bottlerocket_modeled_types::{ApiCommand, BootstrapCommandMode, Identifier};
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::HashMap, convert::Infallible};
+
+/// BootstrapCommandsSettingsV1 represents how a BootstrapCommand lives as a Bottlerocket Setting.
+/// It can be set by either user-data or through Bottlerocket API. Lets take an example
+/// [settings.bootstrap-commands."001-test-bootstrap-commands"]
+/// commands = [["apiclient", "set", "motd=helloworld1"]]
+/// mode = "always"
+/// essential = true
+/// Here, the name (Identifier) is "001-test-bootstrap-commands" and commands, mode and essential
+/// form the BootstrapCommand object.
+/// The Bootstrap Commands are executed in lexicographical order of their names (Identifier).
```

Comment:
This is an atypical comment for this code base. You are specifying the end usage (`settings.bootstrap-commands`) but that isn't really applied or enforced here. I don't think it clarifies this code to introduce the wider context.

Reply:
Had added it on request but I agree it deviates from the repository standard and adds additional context not relevant here.